madmom/ml/nn/layers.py

```diff
@@ -283,16 +290,22 @@ class LSTMLayer(Layer):
         Output gate.
     activation_fn : numpy ufunc, optional
         Activation function.
+    out_init : numpy array, optional
+        Initial state after the output gate.
+    state_init : numpy array, optional
+        Initial state of the internal state.
 
     """
 
     def __init__(self, input_gate, forget_gate, cell, output_gate,
-                 activation_fn=tanh):
+                 activation_fn=tanh, out_init=None, state_init=None):
         self.input_gate = input_gate
         self.forget_gate = forget_gate
         self.cell = cell
         self.output_gate = output_gate
         self.activation_fn = activation_fn
+        self.out_init = out_init
+        self.state_init = state_init
```

Comment:
As above, I'm not really happy with the names here.
Also, for streaming mode, the output of the previous step must be accessible anyways, e.g. as ``self.out``.

For the ``state_init`` I am thinking of moving this as ``init`` to the cell, and store the cell's state there -- this is basically what it is. This needs some change in the ``activate()`` method, but is the cleaner way to solve this (IMO).